README.md

```diff
@@ -12,5 +12,20 @@
 
 ## Usage
 
+1. Put your config file in `conf/config.json`
```

Comment:
Yes, it will still work as a Docker image (b/c the `CMD` is set to `conf.json` by default).

For the CoreOS service template, I've added a check to `script/deploy` that will fail the deployment if `$DEPLOY_ENV` isn't set. It's required in that context.

`$DEPLOY_ENV` isn't used in `start-synapse.sh`, but you probably mean `$CONFIG`. In that case, yes, that is required to be set. I assumed we could fall back on synapse's failure to start there, but I can (and did) add a check in the script itself.